src/utils/time.ts

```diff
@@ -0,0 +1,73 @@
+export type Meridiem = "AM" | "PM";
+
+export type MinuteOrSecond = "minute" | "second";
+
+export interface Time {
+  hour: string;
+  minute: string;
+  second: string;
+}
+
+export type TimeComponents = "hour" | MinuteOrSecond | "meridiem";
```

Comment:
For other components, we've been calling this something like `TimeFocusId` and using dashes for values. `minute-or-second`.